modules/geo-layers/src/tile-layer/tile-2d-header.js

```diff
@@ -17,12 +18,29 @@ export default class Tile2DHeader {
 
     this.onTileLoad = onTileLoad;
     this.onTileError = onTileError;
+
+    this.transformToWorld = null;
+    this._transformToWorldDataCache = [];
   }
 
   get data() {
     return this._isLoaded ? this.content : this._loader;
   }
 
+  get dataInWorldCoordinates() {
+    const data = this._isLoaded ? this.content : this._loader;
+
+    if (this.transformToWorld) {
+      if (!this._transformToWorldDataCache.length) {
+        this._transformToWorldDataCache = this.transformToWorld(data);
```

Comment:
If you pass a promise to `transformToWorld` then `transformToWorld` needs to be an async function.